code/controllers/master_controller.dm

```diff
@@ -507,3 +509,53 @@ datum/controller/game_controller/recover()		//Mostly a placeholder for now.
 				else
 					msg += "\t [varname] = [varval]\n"
 	world.log << msg
+
+
+/datum/controller/game_controller/proc/procedurally_generate()
```

Comment:
It's not hardcoded for snowmap.
I will not use the procedural map generator from TG because it's almost certainly very different than the functionality of this